webapp/channels/src/components/gif_picker/components/gif_picker_items.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useMemo} from 'react';
+import {GiphyFetch} from '@giphy/js-fetch-api';
+import {EmojiVariationsListProps, Grid} from '@giphy/react-components';
+
+const giphyFetch = new GiphyFetch('DxfEk2t4CR8bv4A1kviDBRMP9i1og3Da');
+
+const GUTTER_BETWEEN_GIFS = 8;
+const NUM_OF_GIFS_COLUMNS = 2;
+const GIF_DEFAULT_WIDTH = 350;
+const GIF_MARGIN_ENDS = 12;
+
+interface Props {
+    filter: string;
+    onClick: EmojiVariationsListProps['onGifClick'];
+}
+
+function GifPickerItems(props: Props) {
+    function fetch(offset: number) {
+        // We dont have to throttled the fetching as the library does it for us
+        if (props.filter.length > 0) {
+            return giphyFetch.search(props.filter, {offset, limit: 10});
+        }
+
+        return giphyFetch.trending({offset, limit: 10});
+    }
+
+    const width = useMemo(() => {
+        const picketRoot = document.getElementById('emoji-picker-tabs');
```

Comment:
There's actually three of these components that can be on screen at once between the post/comment textboxes and the edit post textbox. To make sure these are looking at the bounds of the right element, you could either give the `EmojiPickerOverlay`/`EmojiPickerTabs` an ID prop so that all 3 of those can have a different ID or you could pass a prop to this component that's a function that returns a reference to this element. I'd recommend the second approach, but since `emoji-picker-tabs` is a React Bootstrap element, that might be difficult because they probably don't forward props correctly

Reply:
Yes you are correct they didnt forward the props properly and the ref on tabs gave me back the react component of tabs 🤦🏽

Reply:
I guess this is a case where we could use `findDOMNode`, but I'd rather stick with this personally

Reply:
added a wrapper and passed the ref